scripts/west_commands/runners/stlink_gdbserver.py

```diff
@@ -99,21 +99,39 @@ def extload_help(cls) -> str:
     @classmethod
     def do_add_parser(cls, parser: argparse.ArgumentParser):
         # Expose a subset of the ST-LINK GDB server arguments
-        parser.add_argument(
-            "--swd", action='store_true', default=True, help="Enable SWD debug mode"
+        swd_group = parser.add_mutually_exclusive_group()
+        swd_group.add_argument(
+            "--swd",
+            dest='swd',
+            action='store_true',
+            help="Enable SWD debug mode (default: %(default)s)\nUse --no-swd to disable.",
         )
+        swd_group.add_argument("--no-swd", dest='swd', action='store_false', help=argparse.SUPPRESS)
```

Comment:
Please use `action=argparse.BooleanOptionalAction` instead of a mutually exclusive group.

```python
        parser.add_argument(
            "--swd",
            action=argparse.BooleanOptionalAction,
            help="Enable SWD debug mode (default: %(default)s)\nUse --no-swd to disable.",
        )
```